RecoLocalCalo/EcalRecAlgos/src/EcalUncalibRecHitTimingCCAlgo.cc

```diff
@@ -0,0 +1,157 @@
+                                                    const std::vector<double>& amplitudes,
+                                                    const EcalPedestals::Item* aped,
+                                                    const EcalMGPAGainRatio* aGain,
+                                                    const FullSampleVector& fullpulse,
+                                                    EcalUncalibratedRecHit& uncalibRecHit,
+                                                    float errOnTime) const {
+  constexpr unsigned int nsample = EcalDataFrame::MAXSAMPLES;
+
+  double maxamplitude = -std::numeric_limits<double>::max();
+  float pulsenorm = 0.;
+
+  std::vector<float> pedSubSamples(nsample);
+  for (unsigned int iSample = 0; iSample < nsample; iSample++) {
+    const EcalMGPASample& sample = dataFrame.sample(iSample);
+
+    float amplitude = 0.;
+    int gainId = sample.gainId();
+
+    double pedestal = 0.;
+    double gainratio = 1.;
+
+    if (gainId == 0 || gainId == 3) {
+      pedestal = aped->mean_x1;
+      gainratio = aGain->gain6Over1() * aGain->gain12Over6();
+    } else if (gainId == 1) {
+      pedestal = aped->mean_x12;
+      gainratio = 1.;
+    } else if (gainId == 2) {
+      pedestal = aped->mean_x6;
+      gainratio = aGain->gain12Over6();
+    }
+
+    amplitude = (static_cast<float>(sample.adc()) - pedestal) * gainratio;
+
+    if (gainId == 0) {
+      //saturation
+      amplitude = (4095. - pedestal) * gainratio;
+    }
+
+    pedSubSamples[iSample] = amplitude;
+
+    if (amplitude > maxamplitude) {
+      maxamplitude = amplitude;
+    }
+    pulsenorm += fullpulse(iSample);
+  }
+
+  int ipulse = -1;
+  for (auto const& amplit : amplitudes) {
+    ipulse++;
+    int bxp3 = ipulse - 2;
+    int firstsamplet = std::max(0, bxp3);
+    int offset = 7 - bxp3;
+
+    for (unsigned int isample = firstsamplet; isample < nsample; ++isample) {
+      auto const pulse = fullpulse(isample + offset);
+      pedSubSamples[isample] = std::max(0., pedSubSamples[isample] - amplit * pulse / pulsenorm);
+    }
+  }
+
+  // Start of time computation
+  float tStart = startTime_ + GLOBAL_TIME_SHIFT;
+  float tStop = stopTime_ + GLOBAL_TIME_SHIFT;
+  float tM = (tStart + tStop) / 2;
+
+  float distStart, distStop;
+  int counter = 0;
+
+  do {
+    ++counter;
+    distStart = computeCC(pedSubSamples, fullpulse, tStart);
+    distStop = computeCC(pedSubSamples, fullpulse, tStop);
+
+    if (distStart > distStop) {
+      tStop = tM;
+    } else {
+      tStart = tM;
+    }
+    tM = (tStart + tStop) / 2;
+
+  } while (tStop - tStart > targetTimePrecision_ && counter < MAX_NUM_OF_ITERATIONS);
+
+  tM -= GLOBAL_TIME_SHIFT;
+
+  if (counter < MIN_NUM_OF_ITERATIONS || counter > MAX_NUM_OF_ITERATIONS - 1) {
+    tM = TIME_WHEN_NOT_CONVERGING * ecalPh1::Samp_Period;
+    //Negative error means that there was a problem with the CC
+    errOnTime = -targetTimePrecision_ / ecalPh1::Samp_Period;
```

Comment:
BTW, I do not see a value assigned to `errOnTime` for normal conditions.
Is it intentional?